paper/src/main/java/vg/civcraft/mc/civmodcore/world/locations/chunkmeta/ChunkMetaFactory.java

```diff
@@ -1,36 +1,35 @@
 package vg.civcraft.mc.civmodcore.world.locations.chunkmeta;
 
-import java.util.Collection;
-import java.util.HashMap;
-import java.util.Map;
-import java.util.Map.Entry;
-import java.util.TreeMap;
+import java.util.*;
 import java.util.function.Supplier;
 
 public class ChunkMetaFactory {
 
 	private static ChunkMetaFactory instance;
+
 	public static ChunkMetaFactory getInstance() {
-		if (instance == null) {
+		if (instance == null)
 			instance = new ChunkMetaFactory();
-		}
+
 		return instance;
 	}
-	private Map<String, Short> pluginToInternalIdMapping;
 
-	private Map<Short, Supplier<ChunkMeta<?>>> metaInstanciators;
+	private final Map<String, Short> pluginToInternalIdMapping;
+	private final List<ChunkMetaInitializer> initializers;
 
 	private ChunkMetaFactory() {
-		pluginToInternalIdMapping = new HashMap<>();
-		metaInstanciators = new TreeMap<>();
+		this.pluginToInternalIdMapping = new HashMap<>();
+		this.initializers = new ArrayList<>();
 	}
 
-	Collection<Entry<Short, Supplier<ChunkMeta<?>>>> getEmptyChunkFunctions() {
-		return metaInstanciators.entrySet();
+	Collection<ChunkMetaInitializer> getInitializers() {
```

Comment:
On the one hand yes, on the other hand, the app will need to create a new collection on each new chunk load.
Also, this method is visible only inside CMC
Therefore I would prefer to avoid the creation of the new copy, small but anyway boost in performance :)